### PR TITLE
fix(eslint-plugin): [no-inferrable] fix optional param to valid code

### LINKED ISSUE #5340: Bug: no-inferrable-types fixer does not remove `?` when removing the type
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=GYVwdgxgLglg9mABFApgZygCgIYH4BciYIAtgEYoBOiAvIgIwCUiA3ogL4CwAUEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDNFpoAQwBGSWkVKoMkftA7RI4MAF8QyoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

There is a lint error, use the fix command on it.

### Repro Code

```typescript
function test(a?: number = 1) {}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-inferable-types": ["error"],
  },
};
```

### tsconfig

_No response_

### Expected Result

Valid TS Code
```
function test(a = 1) {}
```

### Actual Result

Code that throws ts error

> Parameter cannot have question mark and initializer

```
function test(a? = 1) {}
```

### Additional Info

_No response_

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.30.6` |
| `@typescript-eslint/parser`            | `5.30.6` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.19.0` |
| `node`                                 | `16.16.0` |

### PR DESCRIPTION
**Pull request by Josh-Cena**
## PR Checklist

-   [x] Addresses an existing open issue: fixes #5340
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

(I have a feeling we need more test cases, but I can't think of any edge-cases.)

### CODE REVIEW

File: packages/eslint-plugin/tests/rules/no-inferrable-types.test.ts [hunk 1]
```diff
@@ -156,7 +156,26 @@ class Foo {
 
   invalid: [
     ...invalidTestCases,
-
+    {
+      // This is invalid TS semantic, but it's trivial to make valid anyway
+      code: 'const fn = (a?: number = 5) => {};',
```
Comment (armano2) [inline packages/eslint-plugin/tests/rules/no-inferrable-types.test.ts:161]: not sure if we want to handle this in here, but there is one more use case that is broken.

[playground](https://deploy-preview-5342--typescript-eslint.netlify.app/play/#ts=4.7.2&sourceType=module&code=MYewdgzgLgBAZmGBeGAKAhgfgFwzAVwFsAjAUwCdkYBWASmQD4YBvGAXwG4BYAKF+AA26CBBgAVUtBa8YMAB4BCXARIUq1Xm15A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDNFpoAQwBGSWkVKoMkftA7RI4MAF8QyoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)

```ts
class Test {
  x!: number = 5
}
```

Comment (Josh-Cena) [reply packages/eslint-plugin/tests/rules/no-inferrable-types.test.ts:161]: I'll add definite assignment to this PR, but the `private` one seems tangential to this.

Comment (Josh-Cena) [reply packages/eslint-plugin/tests/rules/no-inferrable-types.test.ts:161]: Also, the `abstract y!: number = 5` is too messed up to be autofixed. The initializer needs to be removed as well. It isn't even reported by the rule under the status quo, so I'll just ignore that.

**Review by bradzacher — APPROVED** [review]
thanks for fixing this!

**Pull request merged**